From 6b85adad56ebb6e81cffe408e78dc6b3bd7fc1e6 Mon Sep 17 00:00:00 2001
From: Stephan Hartmann <stha09@googlemail.com>
Date: Thu, 25 Nov 2021 18:46:12 +0000
Subject: [PATCH] Vulkan: add explicit template specialization for barrierImpl

ImageHelper::barrierImpl with primary command buffer is used in
other translation units (SurfaceVk::acquireNextSwapchainImage()).
Linking fails if compiler has optimized out symbol for primary
command buffer.
---
 .../angle/src/libANGLE/renderer/vulkan/vk_helpers.cpp      | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/third_party/angle/src/libANGLE/renderer/vulkan/vk_helpers.cpp b/third_party/angle/src/libANGLE/renderer/vulkan/vk_helpers.cpp
index b183c7b..1bfddc4 100644
--- a/third_party/angle/src/libANGLE/renderer/vulkan/vk_helpers.cpp
+++ b/third_party/angle/src/libANGLE/renderer/vulkan/vk_helpers.cpp
@@ -5078,6 +5078,13 @@ void ImageHelper::barrierImpl(Context *context,
     mCurrentQueueFamilyIndex = newQueueFamilyIndex;
 }
 
+template void ImageHelper::barrierImpl<rx::vk::priv::CommandBuffer>(
+    Context *context,
+    VkImageAspectFlags aspectMask,
+    ImageLayout newLayout,
+    uint32_t newQueueFamilyIndex,
+    rx::vk::priv::CommandBuffer *commandBuffer);
+
 bool ImageHelper::updateLayoutAndBarrier(Context *context,
                                          VkImageAspectFlags aspectMask,
                                          ImageLayout newLayout,
-- 
2.32.0

